### PR TITLE
 Fix issues for Az.Tools.Installer

### PR DESCRIPTION
**Pull request by msJinLei**
1. Fix bugs in installation and updating
 2. Fix coding formatting issues
 3. Add test cases of installation and updating without Repository

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: tools/Az.Tools.Installer/exports/Update-AzModule.ps1 [hunk 1]
```diff
@@ -91,7 +91,12 @@ function Update-AzModule {
             $findModuleParams.Add('Repository', $Repository)
         }
         $modulesToUpdate = Get-AzModuleFromRemote @findModuleParams
-        $Repository = $modulesToUpdate.Repository | Select-Object -First 1
+        if ($modulesToUpdate) {
+            $Repository = $modulesToUpdate.Repository | Select-Object -First 1
+        }
+        else {
+            $Repository = (Get-PSRepository | Select-Object -Property Name)
```
Comment (isra-fel) [inline tools/Az.Tools.Installer/exports/Update-AzModule.ps1:98]: Get-PSRepository may return more than 1 result. This like also needs "-First 1"?

File: tools/Az.Tools.Installer/internal/Install-AzModule_Default.ps1 [hunk 2]
```diff
@@ -80,7 +80,12 @@ function Install-AzModule_Default {
 
         $modules = @()
         $modules += Get-AzModuleFromRemote @findModuleParams | Sort-Object -Property Name
-        $Repository = $modules.Repository | Select-Object -First 1
+        if ($modules) {
+            $Repository = $modules.Repository | Select-Object -First 1
+        }
+        else {
+            $Repository = (Get-PSRepository | Select-Object -Property Name)
```
Comment (isra-fel) [inline tools/Az.Tools.Installer/internal/Install-AzModule_Default.ps1:87]: same here

File: tools/Az.Tools.Installer/exports/Update-AzModule.ps1 [hunk 3]
```diff
@@ -91,7 +91,12 @@ function Update-AzModule {
             $findModuleParams.Add('Repository', $Repository)
         }
         $modulesToUpdate = Get-AzModuleFromRemote @findModuleParams
-        $Repository = $modulesToUpdate.Repository | Select-Object -First 1
+        if ($modulesToUpdate) {
+            $Repository = $modulesToUpdate.Repository | Select-Object -First 1
+        }
+        else {
+            $Repository = [Array]((Get-PSRepository | Select-Object -Property Name).Name)
```
Comment (wyunchi-ms) [inline tools/Az.Tools.Installer/exports/Update-AzModule.ps1:98]: Why `Repository` is array in `else` condition?

**Comment by msJinLei** [comment]
@isra-fel @wyunchi-ms 
Thanks for the comments.

I checked the codes again and find $Repository in else in `Update-AzModule.ps1` won't be used anymore and so I remove it.
In `Install-AzModule_Default.ps1`,  the only case to use the value assigned in `else` is that when a wrong module name is provided and nothing is to installed but leave a error message. I process $Repository in the message specially to make the code understood more easily.

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**